cli/command/image/build.go

```diff
@@ -181,7 +181,7 @@ func runBuild(dockerCli command.Cli, options buildOptions) error {
 		remote        string
 	)
 
-	if !options.quiet {
+	if !options.quiet && dockerCli.ServerInfo().OSType != "windows" {
 		_, _ = fmt.Fprint(dockerCli.Err(), `WARNING: The legacy builder is in use and will build your image in an inefficient way.
```

Comment:
- The "inefficient" bit feels a bit odd for a warning; perhaps we should mention that it's deprecated, and (the automatic fallback) will be removed in a future release.
- I would not print _both_ messages (or remove the "Required component" from the first warning)

Reply:
> * The "inefficient" bit feels a bit odd for a warning; perhaps we should mention that it's deprecated, and (the automatic fallback) will be removed in a future release.

Done with a deprecated message:

```
$ docker build .
WARNING: Required builder component buildx is missing or broken. To install buildx, see https://docs.docker.com/go/buildx.
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.

Sending build context to Docker daemon ...
...
```

> * I would not print _both_ messages (or remove the "Required component" from the first warning)

I think it's important to keep it to advertise inconsistency in its installation.